components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -255,35 +255,40 @@ tobeDeleted claims (claims came from federated idp as null). If there is a match
                 diagnosticLog.info("Federated user: " + username + " is provisioned locally.");
             }
 
-            if (roles != null) {
-                // Update user with roles
-                List<String> currentRolesList = Arrays.asList(userStoreManager.getRoleListOfUser(username));
-                Collection<String> deletingRoles = retrieveRolesToBeDeleted(realm, currentRolesList, rolesToAdd);
+            boolean includeManuallyAddedLocalRoles = false;
```

Comment:
false is redundant here

Reply:
what if the check fails? Then the includeManuallyAddedLocalRoles will be undefined...

Reply:
booleans are by default false right?